crates/bevy_color/src/hsla.rs

```diff
@@ -23,6 +28,14 @@ pub struct Hsla {
 
 impl StandardColor for Hsla {}
 
+impl_componentwise_add!(Hsla, [hue, saturation, lightness, alpha]);
```

Comment:
For any colour spaces containing an angular channel (e.g., `hue` in `Hsla`), this default form of addition can trivially lead to angle values outside of the principal range [0, 360). This could be a problem for certain colour conversion implementations, as not all of them exclusively pass these channels into trigonometric functions (`sin`, `cos`, etc.).

While a lesser concern, it does also open up the possibility of lost precision as we move further away from the principal range (since IEEE floating point representation loses precision with magnitude).

We _could_ do something fancy, like define a `ColorChannel<const N: usize>` trait which encodes this behaviour, and then implement the component-wise traits based on having 4 channels, or some other work around. But I think the simplest option is to refactor these macros to split the channel list into angular channels, and linear channels.